variables.tf

```diff
@@ -319,6 +319,16 @@ variable "adot_config" {
     log_retention          = optional(number, 14)
     helm_values            = optional(any, null)
     logging_enable         = optional(bool, false)
+    resources = object({
```

Comment:
this requires to set `resources` field every time you pass `adot_config` variable, for example above test `eks-with-adot` will fail to plan/apply. to overcome this just use optional() by having something like this:
```hcl
    resources = optional(object({
      limit = object({
        cpu    = optional(string, "200m")
        memory = optional(string, "200Mi")
      })
      requests = object({
        cpu    = optional(string, "200m")
        memory = optional(string, "200Mi")
      })
      }), {
      limit = {
        cpu    = "200m"
        memory = "200Mi"
      }
      requests = {
        cpu    = "200m"
        memory = "200Mi"
      }
    })
```

Reply:
the `./testa/eks-with-adot` still will fail as `resources` is required here for root module